extensions/2.0/Khronos/KHR_materials_transmission/README.md

```diff
@@ -0,0 +1,147 @@
+
+This extension aims to address the simplest and most common use cases for optical transparency: infinitely-thin materials with no complex scattering (e.g. dynamic scattering) or dispersion. Dealing exclusively with “thin” materials (i.e. materials where only the surface is considered and not the volume) allows many simplifications to be made when calculating things like refraction and absorption.
+
+## Extending Materials
+
+A transparent material is defined by adding the `KHR_materials_transmission` extension to any glTF material. When present, the extension indicates that a material should be rendered as a transparent surface and be blended as defined in the following spec. Alpha coverage and `doubleSided` properties still apply to transparent materials. Note that, as rendering transparent objects presents many difficult-to-solve issues with primitive ordering, this extension does not dictate rendering algorithms.
+
+```json
+materials: [
+  {
+    "extensions": {
+       "KHR_materials_transmission": {
+         "transmissionFactor": 0.8,
+         "transmissionTexture": 0
+       }
+    }
+  }
+]
+```
+
+## Properties 
+
+Only two properties are introduced with this extension and combine to describe a single value; the percentage of light that is transmitted through the surface of the material. These properties work together with the existing properties of the material to define the way light is modified as it passes through the substance. 
+
+|   |Type|Description|Required|
+|---|----|-----------|--------|
+|**transmissionFactor** | `number` | The base percentage of light that is transmitted through the surface.| No, default:`1.0`|
+|**transmissionTexture** | [`textureInfo`] | A greyscale texture that defines the transmission percentage of the surface. This will be multiplied by transmissionFactor. | No |
+
+### transmissionFactor 
+
+The amount of light that is transmitted by the surface rather than diffusely re-emitted. This is a percentage of all the light that penetrates a surface (i.e. isn’t specularly reflected) rather than a percentage of the total light that hits a surface. A value of 1.0 means that 100% of the light that penetrates the surface is transmitted through. This is the default.
+
+<figure>
+  <img src="./figures/ConstantTransmission.png"/>
+<figcaption><em>Defaulting to 1.0 allows making a material transparent by simple defining the extension.</em></figcaption>
+</figure>
+
+### transmissionTexture 
+
+A greyscale texture that defines the amount of light that is transmitted by the surface rather than diffusely re-emitted. A value of 1.0 means that 100% of the light that penetrates the surface (i.e. isn’t specularly reflected) is transmitted through. The value is linear and is multiplied by the transmissionFactor to determine the total transmission value. Note that use of transmission values does not disallow usage of alpha coverage (via `baseColor.a`).
```

Comment:
If alpha coverage is not needed could we have a solution where the transmissionTexture  scalar is taken from baseColor.a?

This would optimize use of samplers and bandwidth for many usecases.

Reply:
I'm a little torn on this.  If there were a good fallback path to be had here, where non-transmission-aware renderers could do the best fallback by finding the transmission values mixed into the alpha coverage channel, maybe that would be a reason to try this?

But for non-fallback behavior, this seems like a problem.  For example Blender keeps these values separate for a reason, and it's possible to use both on the same material.

Reply:
Yes. Use of both alpha blend and transmission seems sketchy to me, but there may be reasons to do so. Use of both alpha *mask* and transmission seems totally reasonable. I think it would be unsafe for this extension to try to alter its behavior based on the alphaMode (and as @emackey points out, Blender avoids that too) so I think the extension is probably correct as written.

Reply:
You _could_ make an argument that the `transmissionTexture`'s single channel should be `a` instead of `r`? But I'm having trouble convincing myself whether that's a good idea... using `baseColorTexture` and `transmissionTexture` with the same texture would cause problems in some engines — it is difficult to really ensure that a 0% alpha channel doesn't have leak-through side effects even when the material is marked as opaque. Especially if that texture is being reused elsewhere _with_ the alpha channel...

Reply:
I think these two things are very different concepts and we shouldn't do anything to confuse them for the user. Besides, using baseColor.a for transmission would require taking the inverse of the value which further confuses things.